File: Neopets Calculator/AutoNPBank.java
//********************************************
// Name: Brian Mascitello
// Title: NPBank.java
// Description: Input Total NP, Output tells Account Type, Current Daily Gains, and NP until +1NP/day Interest
// Time spent: 2 hours
// Date: 9/18/2012
//********************************************

import java.math.RoundingMode;
import java.text.DecimalFormat;
import java.util.Scanner;
//import javax.swing.*;

public class AutoNPBank
{
	public static void main(String[] args)
	{
		DecimalFormat format = new DecimalFormat("#");
		format.setRoundingMode(RoundingMode.UP);
		Scanner scan = new Scanner(System.in);

		double Balance;
		double Rate;
		double Daily;

		/*JFrame frame = new JFrame();

		final int FRAME_WIDTH = 300;
		final int FRAME_HEIGHT = 400;
		frame.setSize(FRAME_WIDTH, FRAME_HEIGHT);
		frame.setTitle("Automatic Neopoint Bank");
		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

		frame.setVisible(true);*/

		System.out.println( "Enter your Current Balance: " );
		Balance = scan.nextInt();
		Rate = 0;

		if ( 0 <= Balance && Balance < 1000 )
		{
			System.out.println( "\nAccount Type: \t \t Junior Saver" );
			Rate = 4.5;
			}

		if ( 1000 <= Balance && Balance < 2500 )
		{
			System.out.println( "\nAccount Type: \t \t Neopian Student" );
			Rate = 5.5;
			}

		if ( 2500 <= Balance && Balance < 5000 )
		{
			System.out.println( "\nAccount Type: \t \t Bronze Saver" );
			Rate = 6.0;
			}

		if ( 5000 <= Balance && Balance < 10000 )
		{
			System.out.println( "\nAccount Type: \t \t Silver Saver" );
			Rate = 6.5;
			}

		if ( 10000 <= Balance && Balance < 25000 )
		{
			System.out.println( "\nAccount Type: \t \t Super Gold Plus" );
			Rate = 7.0;
			}

		if ( 25000 <= Balance && Balance < 50000 )
		{
			System.out.println( "\nAccount Type: \t \t Platinum Extra" );
			Rate = 7.5;
			}

		if ( 50000 <= Balance && Balance < 75000 )
		{
			System.out.println( "\nAccount Type: \t \t Double Platinum" );
			Rate = 8.0;
			}

		if ( 75000 <= Balance && Balance < 100000 )
		{
			System.out.println( "\nAccount Type: \t \t Triple Platinum" );
			Rate = 8.5;
			}

		if ( 100000 <= Balance && Balance < 250000 )
		{
			System.out.println( "\nAccount Type: \t \t Diamond Deposit" );
			Rate = 9.0;
			}

		if ( 250000 <= Balance && Balance < 500000 )
		{
			System.out.println( "\nAccount Type: \t \t Diamond Deposit Plus" );
			Rate = 9.5;
			}

		if ( 500000 <= Balance && Balance < 1000000 )
		{
			System.out.println( "\nAccount Type: \t \t Diamond Deposit Gold" );
			Rate = 10.0;
			}

		if ( 1000000 <= Balance && Balance < 2000000 )
		{
			System.out.println( "\nAccount Type: \t \t Millionaire Platinum" );
			Rate = 10.5;
			}

		if ( 2000000 <= Balance && Balance < 5000000 )
		{
			System.out.println( "\nAccount Type: \t \t Millionaire Double Platinum" );
			Rate = 11.0;
			}

	 	if ( 5000000 <= Balance && Balance < 7500000 )
		{
			System.out.println( "\nAccount Type: \t \t Millionaire Mega-Platinum" );
			Rate = 11.5;
			}

		if ( 7500000 <= Balance && Balance < 10000000 )
		{
			System.out.println( "\nAccount Type: \t \t Neopian Mega-Riches" );
			Rate = 12.0;
			}

		if ( 10000000 <= Balance && Balance < 2147483647 )
		{
			System.out.println( "\nAccount Type: \t \t Ultimate Riches!" );
			Rate = 12.5;
			}

		System.out.println( "Current Balance: \t " + (int)Balance + " NP" );

		System.out.println( "Interest Rate: \t \t " + ( Rate ) + "% per year" );

		System.out.println( "Yearly Interest: \t " + (int)Math.floor( Balance * Rate / 100 ) + " NP" );

		System.out.println( "\nDaily Interest: \t " + format.format( Balance * Rate / 100 / 365 ) + " NP" );

		Daily = Math.ceil( Balance * Rate / 100 / 365 );

		System.out.println( "Next Neopoint/Day: \t " + format.format( (( Daily ) / Rate * 100 * 365 ) - Balance) + " NP" + "\n" ); // + ( Balance % ( 365 / Rate * 100 ))

	}
}

File: Neopets Calculator/AutoNPBank.cpp
/*
Auto Neopoint Bank Calculator C++
Remake of AutoNPBank.java
Made by Brian Mascitello
Date: 11/4/2013
*/

#include <iostream>
#include <stdio.h>

char CommaAdder(int NumberFormat);

double daily, decimal_rate, rate, interest;

int balance, daily_interest, next_neopoint_day, yearly_interest;

using namespace std;

void Account();

int main()
{
	cout << "Auto Neopoint Bank Calculator C++" << endl;
	cout << "Enter your current balance: \t";
	cin >> balance;

	Account();

	decimal_rate = (rate / 100);
	interest = (balance * decimal_rate);
	daily = ceil(interest / 365);

	yearly_interest = (int)floor(interest);
	daily_interest = (int)ceil(interest / 365);
	next_neopoint_day = ceil(((daily / decimal_rate) * 365) - balance);
	
	cout << "\nCurrent Balance: \t " << balance << " NP";
	cout << "\nInterest Rate: \t \t " << rate << "% per year";
	cout << "\nYearly Interest: \t " << yearly_interest << " NP";
	cout << "\n\nDaily Interest: \t " << daily_interest << " NP";
	cout << "\nNext Neopoint/Day: \t " << next_neopoint_day << " NP\n";

	getchar();
	getchar();
}

char CommaAdder(int NumberFormat)
{

	return 'a';
}

void Account()
{
	if ( 0 <= balance && balance < 1000 )
	{
		cout << ( "\nAccount Type: \t \t Junior Saver" );
		rate = 4.5;
	}

	if ( 1000 <= balance && balance < 2500 )
	{
		cout << ( "\nAccount Type: \t \t Neopian Student" );
		rate = 5.5;
	}

	if ( 2500 <= balance && balance < 5000 )
	{
		cout << ( "\nAccount Type: \t \t Bronze Saver" );
		rate = 6.0;
	}

	if ( 5000 <= balance && balance < 10000 )
	{
		cout << ( "\nAccount Type: \t \t Silver Saver" );
		rate = 6.5;
	}

	if ( 10000 <= balance && balance < 25000 )
	{
		cout << ( "\nAccount Type: \t \t Super Gold Plus" );
		rate = 7.0;
	}

	if ( 25000 <= balance && balance < 50000 )
	{
		cout << ( "\nAccount Type: \t \t Platinum Extra" );
		rate = 7.5;
	}

	if ( 50000 <= balance && balance < 75000 )
	{
		cout << ( "\nAccount Type: \t \t Double Platinum" );
		rate = 8.0;
	}

	if ( 75000 <= balance && balance < 100000 )
	{
		cout << ( "\nAccount Type: \t \t Triple Platinum" );
		rate = 8.5;
	}

	if ( 100000 <= balance && balance < 250000 )
	{
		cout << ( "\nAccount Type: \t \t Diamond Deposit" );
		rate = 9.0;
	}

	if ( 250000 <= balance && balance < 500000 )
	{
		cout << ( "\nAccount Type: \t \t Diamond Deposit Plus" );
		rate = 9.5;
	}

	if ( 500000 <= balance && balance < 1000000 )
	{
		cout << ( "\nAccount Type: \t \t Diamond Deposit Gold" );
		rate = 10.0;
	}

	if ( 1000000 <= balance && balance < 2000000 )
	{
		cout << ( "\nAccount Type: \t \t Millionaire Platinum" );
		rate = 10.5;
	}

	if ( 2000000 <= balance && balance < 5000000 )
	{
		cout << ( "\nAccount Type: \t \t Millionaire Double Platinum" );
		rate = 11.0;
	}

	if ( 5000000 <= balance && balance < 7500000 )
	{
		cout << ( "\nAccount Type: \t \t Millionaire Mega-Platinum" );
		rate = 11.5;
	}

	if ( 7500000 <= balance && balance < 10000000 )
	{
		cout << ( "\nAccount Type: \t \t Neopian Mega-Riches" );
		rate = 12.0;
	}

	if ( 10000000 <= balance && balance < 2147483647 )
	{
		cout << ( "\nAccount Type: \t \t Ultimate Riches!" );
		rate = 12.5;
	}
}
